source/OutfitFilter.cpp

```diff
@@ -0,0 +1,135 @@
+/* OutfitFilter.cpp
+Copyright (c) 2022 by RisingLeaf
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "OutfitFilter.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Ship.h"
+
+
+
+OutfitFilter::OutfitFilter(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void OutfitFilter::Load(const DataNode &node)
+{
+	for(const DataNode &child : node)
+	{
+		if(child.Token(0) == "not" && child.Size() >= 3)
+		{
+			if(child.Token(1) == "attribute")
+				for(unsigned int it = 2; it < child.Tokens().size(); ++it)
+					notOutfitAttributes.emplace_back(child.Token(it));
+			else if(child.Token(1) == "tag")
+				for(unsigned int it = 2; it < child.Tokens().size(); ++it)
+					notOutfitTags.emplace_back(child.Token(it));
+		}
+		else if(child.Token(1) == "attribute" && child.Size() >= 2)
+			for(unsigned int it = 1; it < child.Tokens().size(); ++it)
+				outfitAttributes.emplace_back(child.Token(it));
+		else if(child.Token(1) == "tag" && child.Size() >= 2)
+			for(unsigned int it = 1; it < child.Tokens().size(); ++it)
+				outfitTags.emplace_back(child.Token(it));
+	}
```

Comment:
I think you should add smth when the keyword isnt recognized (here, and in the not if)

Reply:
ok

Reply:
dam dam dam daaaa:
done